### PR TITLE
Add ISO and HDD detection

### PR DESCRIPTION
**Pull request by Lothrien**
this patch is the base to work with chd creation. 
It does list getchildren() from ISO or HDD files, other types still not tested.

### CODE REVIEW

File: SabreTools.DatItems/DatItem.cs [hunk 1]
```diff
@@ -207,6 +207,7 @@ public static DatItem Create(BaseFile baseFile)
                     return new Media(baseFile);
 
                 case FileType.CHD:
+                case FileType.ISOArchive:
```
Comment (mnadareski) [inline SabreTools.DatItems/DatItem.cs:210]: `Disk` has a very specific meaning. `Disk` refers exclusively to CHDs that use the `disk` tag in the various DAT formats. Overloading this with `ISO` as well is not expected nor is it something I want to support in this manner. If you wish to create a new tag for ISOs, please do that instead.

File: SabreTools.DatItems/DatItem.cs [hunk 2]
```diff
@@ -610,6 +611,15 @@ public static ConcurrentList<DatItem> Merge(ConcurrentList<DatItem> infiles)
                             saveditem.CopyMachineInformation(file);
                             saveditem.SetName(file.GetName());
                         }
+                        
+                        // add cloneof to parent
+                        // only saveditem add to list (cause dupetype != 0x00)
+                        // used to create parent/child/grand-child chd
+                        // name = parent, cloneof = child, grand-child, more grand-childs (using parent/child)
+                        if (saveditem.Machine.CloneOf == null || !saveditem.Machine.CloneOf.Contains(file.Machine.Name)
```
Comment (mnadareski) [inline SabreTools.DatItems/DatItem.cs:619]: This seems unrelated to the other parts of this PR at least based on the description and title. Please remove these changes or add them to a separate PR.

File: SabreTools.FileTypes/BaseFile.cs [hunk 3]
```diff
@@ -12,6 +13,8 @@
 using SabreTools.Skippers;
 using Compress.ThreadReaders;
 
+using DiscUtils;
```
Comment (mnadareski) [inline SabreTools.FileTypes/BaseFile.cs:16]: Unnecessary newline before this using statement.

File: SabreTools.FileTypes/BaseFile.cs [hunk 4]
```diff
@@ -199,7 +202,10 @@ public BaseFile(string filename, Stream stream, bool getHashes = true)
             // First line of defense is going to be the extension, for better or worse
             if (!HasValidArchiveExtension(input))
                 return outFileType;
-
+            
+            // initialize DiscUtils
```
Comment (mnadareski) [inline SabreTools.FileTypes/BaseFile.cs:206]: Please be consistent with capitalization in comments.

File: SabreTools.FileTypes/BaseFile.cs [hunk 5]
```diff
@@ -467,7 +487,8 @@ private static bool HasValidArchiveExtension(string path)
                 case "tlz":
                 case "zip":
                 case "zipx":
-
+                case "iso":
```
Comment (mnadareski) [inline SabreTools.FileTypes/BaseFile.cs:490]: This is dangerous to have enabled by default. Most people using SabreTools are used to having external hashes for ISOs. This essentially makes it an archive by default.

File: SabreTools.FileTypes/Enums.cs [hunk 6]
```diff
@@ -1298,6 +1300,10 @@ public enum OutputFormat
         TorrentXZ,
         TorrentXZRomba,
         TapeArchive,
+        
+        CHD,
```
Comment (mnadareski) [inline SabreTools.FileTypes/Enums.cs:1304]: If you're really wanting to add this, don't add CHD to this list. Adding CHD to this list means that we'd be _building_ a CHD on output.

File: SabreTools.FileTypes/SabreTools.FileTypes.csproj [hunk 7]
```diff
@@ -15,7 +15,8 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="SharpCompress" Version="0.29.0" />
+    <PackageReference Include="DiscUtils" Version="0.16.13" />
```
Comment (mnadareski) [inline SabreTools.FileTypes/SabreTools.FileTypes.csproj:18]: I know nothing about this package, what does it do? Is there any way that the functionality can be replicated with SharpCompress or in self-written code?

**Review by mnadareski — CHANGES_REQUESTED** [review]

File: SabreTools.FileTypes/Enums.cs (hunk 6 above)
Comment (Lothrien) [reply SabreTools.FileTypes/Enums.cs:1304]: yeah thats what it does...

Comment (mnadareski) [reply SabreTools.FileTypes/Enums.cs:1304]: That's not something that I feel should be in the scope of SabreTools. External tools like `chdman` make more sense to handle that.

File: SabreTools.FileTypes/SabreTools.FileTypes.csproj (hunk 7 above)
Comment (Lothrien) [reply SabreTools.FileTypes/SabreTools.FileTypes.csproj:18]: No, SharpCompress should stay. its complete different packages and is only for disc and hard disk type (chd, ntfs etc.). First idea is to read an iso (as archive or as whole file) and make an output as chd file. chd files also can be used for hard disk copies, so with discutils you can read every hard disk format and bring it into chd format. As code is designed its also possible to transform it into an iso file. At the moment Id only tested iso or chd packed in 7z files. They were also transformed expectedly. The tranform of chds back to iso or other formats does not work yet.

File: SabreTools.DatItems/DatItem.cs (hunk 1 above)
Comment (Lothrien) [reply SabreTools.DatItems/DatItem.cs:210]: see later, transform into chd is what the basic idea is, so I used Disk for it. Transform into ISO is not used yet...

**Comment by mnadareski** [comment]
Both a combination of the one-way nature of the changes and the fact that I think that external tools such as `chdman` will do a far better job than SabreTools at this, I do not feel comfortable adding this functionality.

The way that this is written does not appear to be well-gated behind any flags so this changes how it will work for all users. Most current users of SabreTools are used to external hashes of ISO and HDD images and _only_ the standard set of internal hashes for all 5 CHD versions.
